es-core/src/guis/GuiInputConfig.cpp

```diff
@@ -342,27 +342,38 @@ bool GuiInputConfig::filterTrigger(Input input, InputConfig* config, int inputId
 #if defined(__linux__)
 	// on Linux, some gamepads return both an analog axis and a digital button for the trigger;
 	// we want the analog axis only, so this function removes the button press event
-
-	if((
-	  // match PlayStation joystick with 6 axes only
-	  strstr(config->getDeviceName().c_str(), "PLAYSTATION") != NULL
-	  || strstr(config->getDeviceName().c_str(), "PS3 Ga") != NULL
-	  || strstr(config->getDeviceName().c_str(), "PS(R) Ga") != NULL
-	  // BigBen kid's PS3 gamepad 146b:0902, matched on SDL GUID because its name "Bigben Interactive Bigben Game Pad" may be too generic
-	  || strcmp(config->getDeviceGUIDString().c_str(), "030000006b1400000209000011010000") == 0
-	  ) && InputManager::getInstance()->getAxisCountByDevice(config->getDeviceId()) == 6)
+	bool isPlaystation = (
+		// match PlayStation joystick with 6 axes only
+		strstr(config->getDeviceName().c_str(), "PLAYSTATION") != NULL
+		|| strstr(config->getDeviceName().c_str(), "Sony Interactive") != NULL // Official dualshock 4
+		|| strstr(config->getDeviceName().c_str(), "PS3 Ga") != NULL
+		|| strstr(config->getDeviceName().c_str(), "PS(R) Ga") != NULL
+		// BigBen kid's PS3 gamepad 146b:0902, matched on SDL GUID because its name "Bigben Interactive Bigben Game Pad" may be too generic
+		|| strcmp(config->getDeviceGUIDString().c_str(), "030000006b1400000209000011010000") == 0
+	);
+	bool isAnbernic = (
+		strcmp(config->getDeviceGUIDString().c_str(), "03004ab1020500000913000010010000") == 0 // Anbernic RG P01 has same issue
+	);
+
+	if((isPlaystation || isAnbernic) 
+		&& InputManager::getInstance()->getAxisCountByDevice(config->getDeviceId()) == 6
+		|| input.type == TYPE_BUTTON)
```

Comment:
This doesn't sound right at first glance? This will always trigger if input.type == TYPE_BUTTON, even if it isn't playstation or anbernic, only then to be led to a condition where it'll always fail if it isn't either playstation or anbernic. Furthermore, if it isn't a button but it's playstation it'll still be true, and then it'll run the internal if when it's not a button, going against the previous logic.

Can we make the logic be comparable to the previous one - and also more readable - by moving the input.type == TYPE_BUTTON to the if statement inside this IF statement (or alternatively at least just make it && rather than || )?

Reply:
Yeah sorry about that, I should have that fixed in the new commit